src/aiperf/gpu_telemetry/pynvml_collector.py

```
@@ -0,0 +1,452 @@
+    async def _collect_and_process_metrics(self) -> None:
+        """Collect metrics from all GPUs and send via callback.
+
+        Gathers current metrics from all discovered GPUs using NVML APIs,
+        converts them to TelemetryRecord objects, and delivers via callback.
+        Uses asyncio.to_thread() to avoid blocking the event loop with NVML calls.
+        """
+        try:
+            records = await asyncio.to_thread(self._collect_gpu_metrics)
+            if records and self._record_callback:
+                await self._record_callback(records, self.id)
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)